src/compiler/parser.ts

```diff
@@ -1227,7 +1226,9 @@ namespace ts {
 
         function parsePropertyNameWorker(allowComputedPropertyNames: boolean): PropertyName {
             if (token() === SyntaxKind.StringLiteral || token() === SyntaxKind.NumericLiteral) {
-                return <StringLiteral | NumericLiteral>parseLiteralNode(/*internName*/ true);
+                const node = <StringLiteral | NumericLiteral>parseLiteralNode();
+                internIdentifier(node.text);

```

Comment:
I'll change it, but according to my knowledge of strings in V8 (which, admitted, is gleaned from stackoverflow answers), reassignment of the value from the map isn't required to make the runtime intern the string.

It does this:
```ts
let identifier = identifiers.get(text);
if (identifier === undefined) {
    identifiers.set(text, identifier = text);
}
return identifier;
```
Its primary use in our code is setting up the `identifiers` list which is used in the emitter and declaration emitter for knowing what identifiers it needs to avoid to generate names. Secondarily, this will cause the string to be interned in V8, since it's used as a property name (or map key in ES6) - a reassignment of the version from the map should not be required (since the same string primitive that's passed in is what gets interned (thus is already assigned to the object) or is already interned if already in the map).

TBQH I feel like the function should be renamed, since interning is not its primary use. Could probably reduce the space used by it a little bit by making it a `Map<true>`, too.